Data-Structure-Linked-Lists.md

```diff
@@ -0,0 +1,403 @@
+# Linked list
+
+ A linked list is visually more like a garland. We call every flower-y stuff on this particular garland to be a node. And each of the node points to the next node in this list as well as it has data (here it is type of flower).
+
+**Types:**
+
+1. Singly Linked List
+
+    Singly linked lists contain nodes which have a `data` field as well as a `next` field, which points to the next node in the sequence. Operations that can be performed on singly linked lists are insertion, deletion and traversal.
+
```

Comment:
To a new reader, it would be confusing because they won't be able to relate why would you need all these, as in, in what practical situation you need a circular list, or a doubly linked list.

You can provide examples of where in a practical situation one would be better over other. For instance, the browser back and front button might internally use a doubly linked list to make both of then O(1) operation for page retrieval.

Similarly, a circular list could be used to create an image slider in a website. It's not that they are used, it's that internally they _might_ be used.